Frontend/ui-customer/src/Screens/ZooTicketBookingFlow/TicketBookingScreen/View.purs

```diff
@@ -293,244 +365,696 @@ feeBreakUpView state push services =
+                          else
+                              if (endTime /= "") then do
+                                if (now < endTime) then find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                                else Nothing 
+                              else Nothing
+                        else Nothing
+                      _ ->  if (endTime /= "") then do
+                              if (now < endTime) then find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+                              else Nothing 
+                            else Nothing
+                else find (\bh -> bh.bhourId == sti.bhourId && selOperationalDay `elem` bh.operationalDays) bhs
+        Just bhId -> find (\bh -> bh.bhourId == bhId && selOperationalDay `elem` bh.operationalDays) bhs
+
+placeClosed :: Maybe TicketPlaceResp -> Boolean
+placeClosed mbPlaceInfo = do
+  case mbPlaceInfo of
+    Nothing -> false
+    Just (TicketPlaceResp pInfo) -> do
+      let currentTime = convertUTCtoISC (getCurrentUTC "") "HH:mm:ss"
+      case pInfo.closeTimings of
+        Nothing -> false
+        Just closeTime -> case pInfo.openTimings of
+            Nothing -> false
+            Just startTime -> do
+              if (convertUTCTimeToISTTimeinHHMMSS startTime) <= currentTime && currentTime <= (convertUTCTimeToISTTimeinHHMMSS closeTime) then false else true
+
+placeClosedToday :: Maybe TicketPlaceResp -> String -> Boolean
+placeClosedToday mbPlaceInfo dateOfVisit = do
+  case mbPlaceInfo of
+    Nothing -> false
+    Just (TicketPlaceResp pInfo) -> do
+      let currentTime = convertUTCtoISC (getCurrentUTC "") "HH:mm:ss"
+          currentDate = convertUTCtoISC (getCurrentUTC "") "YYYY-MM-DD"
+      if currentDate == dateOfVisit then
+        case pInfo.closeTimings of
+          Nothing -> false
+          Just closeTime -> do
+            if currentTime <= (convertUTCTimeToISTTimeinHHMMSS closeTime) then false else true
+      else false
+
+individualTicketBHView :: forall w. (Action -> Effect Unit) -> ST.TicketBookingScreenState -> Maybe ST.TicketBusinessHoursOptionData -> ST.Ticket -> PrestoDOM (Effect Unit) w
+individualTicketBHView push state valBH ticket =
+  PrestoAnim.animationSet [
+    Anim.translateInYAnim translateYAnimConfig { duration = 3000 , fromY = -10 , toY = 0}
+  ] $ 
+  linearLayout
+  [ height WRAP_CONTENT
+  , width MATCH_PARENT
+  , orientation VERTICAL
+  ] $ [
+  ] 
+    <> (case valBH of 
+          Nothing -> []
+          Just bh -> if DA.length bh.categories > 1 then [selectDestinationView push state bh.categories] else (map (incrementDecrementView push state) (convertBHToPeopleCategory bh.categories)))
+    <> (if DA.length ticket.slot > 0 then [timeSlotView push state ticket.ticketID ticket.selectedBHid ticket.slot] else [])
+  where 
+    convertBHToPeopleCategory categories = do
```

Comment:
convertBHToPeopleCategory is defined in 2places , Can we make a function and use

Reply:
done